x-pack/filebeat/input/httpjson/request.go

```diff
@@ -42,6 +47,7 @@ func (c *httpClient) do(stdCtx context.Context, req *http.Request) (*http.Respon
 		return nil, fmt.Errorf("failed to execute http client.Do: %w", err)
 	}
 	defer resp.Body.Close()
+	c.tracer.trace(req, resp)
```

Comment:
I think the request might need to be dumped before it is executed in order to access the body before it is drained.

Reply:
If you add this and run the tests you will see a log file with the request and response logging in "test.log".

```diff
diff --git a/x-pack/filebeat/input/httpjson/input_test.go b/x-pack/filebeat/input/httpjson/input_test.go
index 69617d82a5..d48f8d5ceb 100644
--- a/x-pack/filebeat/input/httpjson/input_test.go
+++ b/x-pack/filebeat/input/httpjson/input_test.go
@@ -583,6 +583,7 @@ func TestInput(t *testing.T) {
        for _, testCase := range testCases {
                tc := testCase
                t.Run(tc.name, func(t *testing.T) {
+                       tc.baseConfig["request.tracer.filename"] = "test.log"
                        tc.setupServer(t, tc.handler, tc.baseConfig)
 
                        cfg := conf.MustNewConfigFrom(tc.baseConfig)
```